fe/src/main/java/org/apache/doris/load/loadv2/BitmapValue.java

```diff
@@ -0,0 +1,374 @@
+                        this.bitmapType = BITMAP_VALUE;
+                        break;
+                }
+                break;
+        }
+    }
+
+    public boolean equals(BitmapValue other) {
+        boolean ret = false;
+        switch (bitmapType) {
+            case EMPTY:
+                switch (other.bitmapType) {
+                    case EMPTY:
+                        ret = true;
+                        break;
+                    case SINGLE_VALUE:
+                    case BITMAP_VALUE:
+                        ret = false;
+                        break;
+                }
+                break;
+            case SINGLE_VALUE:
+                switch (other.bitmapType) {
+                    case EMPTY:
+                    case BITMAP_VALUE:
+                        break;
+                    case SINGLE_VALUE:
+                        ret = this.singleValue == other.singleValue;
+                        break;
+                }
+                break;
+            case BITMAP_VALUE:
+                switch (other.bitmapType) {
+                    case EMPTY:
+                    case SINGLE_VALUE:
+                        ret = false;
+                        break;
+                    case BITMAP_VALUE:
+                        ret = bitmap.equals(other.bitmap);
+                        break;
+                }
+                break;
+        }
+        return ret;
+    }
+
+    /**
+     *  usage note:
+     *      now getSizeInBytes is different from be' impl
+     *      The reason is that java's roaring didn't implement method #shrinkToFit but be's getSizeInBytes need it
+     *      Implementing java's shrinkToFit means refactor roaring whose fields are all unaccess in Doris Fe's package
+     *      That would be an another big project
+     */
+    // TODO(wb): keep getSizeInBytes consistent with be and refactor roaring
+    public long getSizeInBytes() {
+        long size = 0;
+        switch (bitmapType) {
+            case EMPTY:
+                size = 1;
+                break;
+            case SINGLE_VALUE:
+                if (isLongValue32bitEnough(singleValue)) {
+                    size = 1 + 4;
+                } else {
+                    size = 1 + 8;
+                }
+                break;
+            case BITMAP_VALUE:
+                size = 1 + bitmap.getSizeInBytes();
+        }
+        return size;
+    }
+
+    @Override
+    public String toString() {
+        String toStringStr = "{}";
+        switch (bitmapType) {
+            case EMPTY:
+                break;
+            case SINGLE_VALUE:
+                toStringStr = String.format("{%s}", singleValue);
+                break;
+            case BITMAP_VALUE:
+                toStringStr = this.bitmap.toString();
+                break;
+        }
+        return toStringStr;
+    }
+
+    public void clear() {
+        this.bitmapType = EMPTY;
+        this.bitmap = null;
```

Comment:
Would better change the singleValue to a negative value.

Reply:
👌